addons/website_sale/models/product_template.py

```diff
@@ -284,10 +284,9 @@ def _get_sales_prices(self, website):
                     pricelist_price, currency, product_taxes, taxes, self, website=website,
                 ),
             }
-            if pricelist_rule_id:  # If a rule was applied, there might be a discount
-                # For ecommerce flows, the base price is always the product sales price
-                # which can be computed by calling `_compute_base_price` without a pricelist rule
-                pricelist_base_price = template.env['product.pricelist.item']._compute_base_price(
+            pricelist_item = template.env['product.pricelist.item'].browse(pricelist_rule_id)
+            if pricelist_rule_id and pricelist_item.compute_price == 'percentage':
+                pricelist_base_price = pricelist_item._compute_base_price(
```

Comment:
Was this change intentional? `_compute_base_price` was called on an empty recordset before.

Reply:
Yes, previously the base price on ecommerce was always the sales price, whereas in sales it was the pricelist base price.
As we want to harmonise ecommerce and sales behavior, we'll compute the strikethrough price as the pricelist rule base price, and not always the sales price anymore.

Reply:
Hmm, should we do the same below then? (in `_get_additionnal_combination_info`)

Reply:
Indeed, I thought I did it but it seems I forgot :D